### ISSUE TITLE
How to train a deeplab model with 600*600 input image but 500*500 output mask

### ISSUE THREAD

**Issue by liuchangf** [tensorflow/models#7148]
Hello There:

What I want to do is train deeplab V3+ model with different input and output size , e.g. input size 600X600 output mask size 500X500 ,output mask is central based as following ,blue box is input image ,red is output mask. Any idea will be appreciated.
![Screenshot from 2019-07-04 14-12-09](https://user-images.githubusercontent.com/31433358/60643506-ddb4a600-9e65-11e9-88c5-af2816d3cda5.png)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by hbarovertwo** [comment]
Luckily your training set will be easy to create. Just create grayscale label images for as many training images as you want to have where the ratio of central mask is ~0.694 compared to overall image. Make sure to label the outer region as 0 (background) and the inner portion as 1 on your label png files. This means accessing and editing mask image pixel values in grayscale mode. Good luck!

**Comment by liuchangf** [comment]
Thanks for your reply , 
But where I can set the ratio of central mask ?
I can create the train dataset by myself, but deeplab will raise error if input size is different from output mask , so I want to know where I can set the parameter in deeplab ?

**Comment by hbarovertwo** [comment]
I may have misunderstood you at first. 
You want the mask image that deeplab predicts to be resized at the end, right? 
For doing inference with deeplab you have to model a script after the example notebook they give - This would be where you can do any resizing, etc you want to the predicted mask. Your input images (original + mask image) need to be the same size. But you are free to resize the outputs deeplab predicts once you are done training and using the frozen model for inference.

**Comment by liuchangf** [comment]
Thank you for kindly reply.
But I want to modify the deeplab source code which make inference size different from input size (e.g. input size 600X600 output mask size 500X500) .

**Comment by zxwxz** [comment]
The final tensor of deeplab using bilinear resize to original input size
You can modify below part for your request

`for output in sorted(outputs_to_scales_to_logits):
    scales_to_logits = outputs_to_scales_to_logits[output]
    logits = tf.image.resize_bilinear(
        scales_to_logits[MERGED_LOGITS_SCOPE],
        tf.shape(images)[1:3],
        align_corners=False)
    predictions[output] = tf.argmax(logits, 3)`

**Comment by liuchangf** [comment]
Thank you for kindly reply. I will try it.
